src/DockManagerCore/Utilities/DockingUtils.cs

```diff
@@ -0,0 +1,267 @@
+﻿/*
+ * Morgan Stanley makes this available to you under the Apache License,
+ * Version 2.0 (the "License"). You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership. Unless required by applicable law or agreed
+ * to in writing, software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+using System;
+using System.Drawing;
+using System.Windows;
+using System.Windows.Forms;
+using System.Windows.Interop;
+using DockManagerCore.Services;
+using Application = System.Windows.Application;
+
+namespace DockManagerCore.Utilities
+{
+    static class DockingUtils
+    {
+        private static readonly DockingPlaceholder VerticalDock = new DockingPlaceholder { IsGlobalDocker = true };
+        private static readonly DockingPlaceholder HorizontalDock = new DockingPlaceholder { IsGlobalDocker = true };
+
+        public static FloatingWindow FindWindow(string wid)
+        {
+            WindowCollection windows = Application.Current.Windows;
+            foreach (Window w in windows)
+            {
+                if (w is FloatingWindow && w.Uid.Equals(wid))
+                {
+                    return w as FloatingWindow;
```

Comment:
This code can be simplified:
`if (w is FloatingWindow floatingWindow && ...
   return floatingWindow;`